lib/ProkkaAnnotation/ProkkaAnnotationImpl.py

```diff
@@ -94,13 +99,65 @@ def annotate(self, ctx, params):
 
         self.config['ctx'] = ctx
         prokka_runner = ProkkaUtils(self.config)
-
+        report_message = ''
```

Comment:
I think it would be cleaner
1) Move this logic to the utils file
2) Break up the return processing and report creation into smaller functions to instead of having one large function to deal with all of the cases. (unfortunately a lot of the older codebase has very long functions)